### ISSUE TITLE
is there any ignore_index ability in the loss calculation?

### ISSUE THREAD

**Issue by exnx** [EleutherAI/gpt-neox#1193]
Is there a way to incorporate an `ignore_index` ability like cross_entropy in pytorch?  Right now the default is a sequence packing so I guess taking the loss across the whole sequence makes sense (not much padding then).  I added an ability to remove the sequence packing, and just has a padding to fill up the context in the samples.  But I'd like to ignore those in the loss calulation. 

I was curious if anybody knew of this feature or has implemented it themselves?  Thanks!

**Comment by haileyschoelkopf** [comment]
https://github.com/EleutherAI/gpt-neox/blob/7267a74b4893529d87a084a1d8001126aa669a28/megatron/utils.py#L102-L104

the `loss_mask` can be set to 0 to ignore loss for certain token positions! (I can't recall if it is right-shifted or not compared to input token ids though.)

**Comment by Quentin-Anthony** [comment]
@exnx -- Please reopen if this doesn't answer your question! :)
